src/thunderbots/software/radio_communication/mrf/dongle.cpp

```diff
@@ -283,25 +283,40 @@ void MRFDongle::handle_mdrs(AsyncOperation<void> &op)
 
 void MRFDongle::handle_message(AsyncOperation<void> &, USB::BulkInTransfer &transfer)
 {
+    bool to_beep = false;
+
     transfer.result();
 
     // Only handle if there are more than 2 bytes in the transfer.
     if (transfer.size() > 2)
     {
         unsigned int robot = transfer.data()[0];
-        annunciator.handle_robot_message(robot, transfer.data() + 1, transfer.size() - 3,
-                                         transfer.data()[transfer.size() - 2],
-                                         transfer.data()[transfer.size() - 1]);
+        to_beep            = annunciator.handle_robot_message(
+            robot, transfer.data() + 1, transfer.size() - 3,
+            transfer.data()[transfer.size() - 2], transfer.data()[transfer.size() - 1]);
     }
     transfer.submit();
+
+    // If there are new messages since the last update, beep.
+    if (to_beep)
+    {
+        beep(ANNUNCIATOR_BEEP_LENGTH_MILLISECONDS);
+    }
```

Comment:
Wouldn't this mean the dongle is continuously beeping? Or am I misunderstanding when we'd receive a message?

Reply:
Nope, it only beeps when there are new messages compared to the last status update. `handle_robot_message` performs this check